.github/workflows/build.yaml

```diff
@@ -46,6 +46,11 @@ jobs:
           distribution: 'zulu'
           java-version: 21
 
+      - uses: gradle/gradle-build-action@v2
+        # Don't spend more than 5m rehydrating the cache. Otherwise just move on.
+        timeout-minutes: 5
```

Comment:
This may be a little high. Start without to see what numbers you get. We were seeing 20+ minutes on Redwood in rare cases.